rotkehlchen/exchanges/binance.py

```diff
@@ -534,56 +534,88 @@ def _query_lending_balances(
         May raise:
         - RemoteError
         """
-        data = self.api_query_dict('sapi', 'lending/union/account')
-        positions = data.get('positionAmountVos', None)
-        if positions is None:
+        all_positions = []
+        try:
+            timestamp = ts_now_in_ms()
+            current = 1
+            while True:  # query all flexible positions
+                if len(positions := self.api_query_list(
+                    api_type='sapi',
+                    method='simple-earn/flexible/position',
+                    options={
+                        'timestamp': timestamp,
+                        'current': current,
+                        'size': BINANCE_SIMPLE_EARN_HISTORY_PAGE_SIZE,
+                    },
+                )) > 0:
+                    all_positions.append(('totalAmount', positions))
+                    current += 1
+                else:
+                    break
+
+            current = 1
+            while True:  # query all locked position
+                if len(positions := self.api_query_list(
+                    api_type='sapi',
+                    method='simple-earn/locked/position',
+                    options={
+                        'timestamp': timestamp,
+                        'current': current,
+                        'size': BINANCE_SIMPLE_EARN_HISTORY_PAGE_SIZE,
+                    },
+                )) > 0:
+                    all_positions.append(('amount', positions))
+                    current += 1
+                else:
+                    break
+        except RemoteError as e:
             raise RemoteError(
-                f'Could not find key positionAmountVos in lending account data '
-                f'{data} returned by {self.name}.',
-            )
+                f'Could not query simple earn account balances at {timestamp}. {e!s}',
+            ) from e
 
-        for entry in positions:
-            try:
-                amount = deserialize_asset_amount(entry['amount'])
-                if amount == ZERO:
+        for amount_key, positions in all_positions:
+            for entry in positions:
+                try:
+                    amount = deserialize_asset_amount(entry[amount_key])
+                    if amount == ZERO:
+                        continue
+
+                    asset = asset_from_binance(entry['asset'])
+                except UnsupportedAsset as e:
+                    self.msg_aggregator.add_warning(
+                        f'Found unsupported {self.name} asset {e.identifier}. '
+                        f'Ignoring its lending balance query.',
+                    )
+                    continue
+                except UnknownAsset as e:
+                    self.msg_aggregator.add_warning(
+                        f'Found unknown {self.name} asset {e.identifier}. '
+                        f'Ignoring its lending balance query.',
+                    )
+                    continue
```

Comment:
Like you did elsewhere, I think those can be combined with an extra if check in the case of the exception

```suggestion
combine them
```